wts/auth_plugins/__init__.py

```diff
@@ -3,17 +3,7 @@
 import flask
 
 
-def setup_plugins(app):
-    app.auth_plugins = []
-    for plugin in app.config["AUTH_PLUGINS"]:
-        if plugin == "default":
-            app.auth_plugins.append(DefaultPlugin())
-        elif plugin == "k8s":
-            app.auth_plugins.append(K8SPlugin())
-
-
 def find_user():
-    for plugin in flask.current_app.auth_plugins:
-        user = plugin.find_user()
-        if user:
-            return user
+    if flask.request.headers.get("Authorization"):
+        return DefaultPlugin().find_user()
+    return K8SPlugin().find_user()
```

Comment:
we may want to keep the configuration and plugin setup but change it to only handle situations where a token is not provided. e.g. some alternative auth provider besides fence (in this case, the only one we'd have right now is k8s but maaaybe there's a case to be made for something else in the future?) maybe this is YAGNI, but the codes already basically there, we could just always setup "default" and allow cfg setup of "k8s". Then here we iterate over the plugins AFTER we check if there's an Authorization header.